command/agent/config_test.go

```diff
@@ -114,6 +109,11 @@ func TestConfig_Merge(t *testing.T) {
 			NumSchedulers:     2,
 			EnabledSchedulers: []string{structs.JobTypeBatch},
 			NodeGCThreshold:   "12h",
+			RejoinAfterLeave:  true,
```

Comment:
The only tests that use these options are TestConfig_Merge, TestConfig_LoadConfigString, and TestRetryJoin. Is there a specific test case that you are thinking of?